packages/flare/tests/bin/test_ingest_events.py

```diff
@@ -30,58 +34,54 @@
 from cron_job_ingest_events import save_last_ingested_tenant_id
 
 
-def test_get_collection_value_expect_none() -> None:
-    kvstore = MagicMock()
+def test_get_collection_value_expect_none(kvstore: FakeKVStoreCollections) -> None:
     assert get_collection_value(kvstore=kvstore, key="some_key") is None
 
 
-def test_get_collection_value_expect_result() -> None:
-    kvstore = MagicMock()
-    kvstore.__contains__.side_effect = lambda x: x == KV_COLLECTION_NAME
-    kvstore[KV_COLLECTION_NAME].data.query.return_value = [
-        {
-            "_key": "some_key",
-            "value": "some_value",
-        },
-    ]
+def test_get_collection_value_expect_result(kvstore: FakeKVStoreCollections) -> None:
+    kvstore.create(name=KV_COLLECTION_NAME, fields={})
+    kvstore[KV_COLLECTION_NAME].data.insert(
+        json.dumps({"_key": "some_key", "value": "some_value"})
+    )
 
     assert get_collection_value(kvstore=kvstore, key="some_key") == "some_value"
 
 
-def test_save_collection_value_expect_insert() -> None:
+def test_save_collection_value_expect_insert(kvstore: FakeKVStoreCollections) -> None:
     key = "some_key"
     value = "some_value"
-    kvstore = MagicMock()
     save_collection_value(kvstore=kvstore, key=key, value=value)
-    kvstore[KV_COLLECTION_NAME].data.insert.assert_called_once_with(
-        json.dumps({"_key": key, "value": value})
-    )
+    assert kvstore[KV_COLLECTION_NAME].data.query() == [{"_key": key, "value": value}]
```

Comment:
Instead of testing the fake KV store, wouldn't it be better to test that `get_collection_value` returns the proper value? This test and the one before could be merged together to test the save + getter instead

Reply:
These tests: `test_get_collection_value_expect_none` & `test_get_collection_value_expect_result` do exectly that. There's no need to use it in every other test. It wouldn't hurt to use it in every other test, but that's not the focus of these other tests. In this test in particular, `save_collection_value()` is the focus and I'd personally prefer to go directly to the source for the assertion rather than using the wrapper around it. ¯\_(ツ)_/¯, but I'm easy. I can be swayed if someone makes a good argument.

Reply:
Yeah that's fair, the method and the direct query does the same thing so it's actually just a nit